packages/exporter-trace-otlp-http/src/OTLPExporterBase.ts

```diff
@@ -102,6 +110,26 @@ export abstract class OTLPExporterBase<
     });
   }
 
+  private _configureTimeout(timeoutMillis: number | undefined): number {
+    let defaultTimeout = getEnv().OTEL_EXPORTER_OTLP_TRACES_TIMEOUT;
+
+    if ( defaultTimeout === null) {
```

Comment:
the default cannot be null right ? its 10000 by default

Reply:
That's correct. So if `OTEL_EXPORTER_OTLP_TRACES_TIMEOUT` takes priority over the generic timeout `OTEL_EXPORTER_OTLP_TIMEOUT` environment variable, would it make sense to just take out the mention of `OTEL_EXPORTER_OTLP_TIMEOUT` here? 

I'm having a hard time understanding the use of `OTEL_EXPORTER_OTLP_TIMEOUT` in the otlp trace exporters?

Reply:
I think we are meant to check if `OTEL_EXPORTER_OTLP_TRACES_TIMEOUT` is defined, if so yes use it then check `OTEL_EXPORTER_OTLP_TIMEOUT` and finally fallback to the default value

Reply:
Got it. 

I added some logic around this selection process [here](https://github.com/open-telemetry/opentelemetry-js/blob/da3211f44bf32d4568fe6ceee660b4996d897eed/packages/exporter-trace-otlp-http/src/OTLPExporterBase.ts#L123). 

If a user defined `OTEL_EXPORTER_OTLP_TRACES_TIMEOUT` then it will use that first. 

Otherwise it'll check if the user defined `OTEL_EXPORTER_OTLP_TIMEOUT` and use that instead. 

And if neither of those are defined then it'll use the default `OTEL_EXPORTER_OTLP_TRACES_TIMEOUT` as specified in the exporter environments which is 10000.

Reply:
Another question:

Is the exporter timeout allowed to be set via config settings when creating a new `OTLPTraceExporter`?

Example: 

```js
const collectorOptions = {
  timeoutMillis: 15000
};

const traceExporter = new OTLPTraceExporter(collectorOptions);
```

That is how I have it setup now that the exporter will use `timeoutMillis` in the `collectorOptions` if it's provided. If there is no `timeoutMillis` provided in the `collectorOptions` then the exporter will check the user defined environment variables (`OTEL_EXPORTER_OTLP_TRACES_TIMEOUT` and `OTEL_EXPORTER_OTLP_TIMEOUT`).

I assumed that if the `BatchSpanProcessor` timeout is set that way (through the config settings), then it would be a similar process for setting timeout in the otlp trace exporter. But please correct me if I'm wrong.

Reply:
I guess if thats not spec'ed its up to us, i agree that we should either be able to control options via the environment or the code

Reply:
Sounds good. I'll add some information to the readme for each otlp trace exporter about setting this timeout via code.